code/modules/mob/living/silicon/robot/component.dm

```diff
@@ -107,6 +117,13 @@
 	external_type = /obj/item/robot_parts/robot_component/diagnosis_unit
 	max_damage = 30
 
+//NEW INTEGRATED LIGHT COMPONENT
+/datum/robot_component/integrated_light
+	name = "integrated light"
+	energy_consumption = 4
```

Comment:
Powerful light + energy consumption on par with using a module slot.

It's easily toggled on and off. I don't see the problem here.